src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -208,6 +210,19 @@ export class KeyStrokeHandler {
             }
         }
     }
+
+    async executeWhileKeyStrokeHandlerOff(runnable: () => Promise<void>) {
+        this._on = false
+        try {
+            await runnable()
+        } finally {
+            this._on = true
+        }
+    }
```

Comment:
nit: cheap generalizations are usually a good thing to do even if not necessarily needed immediately
```suggestion
    async executeWhileKeyStrokeHandlerOff<T>(runnable: () => Promise<T>): Promise<T> {
        this._on = false
        try {
            return await runnable()
        } finally {
            this._on = true
        }
    }
```

Reply:
updated